airflow-core/src/airflow/api_fastapi/core_api/security.py

```diff
@@ -289,6 +378,43 @@ def inner(
     return inner
 
 
+def requires_access_variable_bulk() -> Callable[[BulkBody[VariableBody], BaseUser], None]:
+    def inner(
+        request: BulkBody[VariableBody],
+        user: GetUserDep,
+    ) -> None:
+        requests: list[IsAuthorizedVariableRequest] = []
+        for action in request.actions:
+            if action.action == BulkAction.DELETE:
+                requests.extend(
+                    [
+                        {
+                            "method": MAP_BULK_ACTION_TO_AUTH_METHOD[action.action],
+                            "details": VariableDetails(key=cast("str", variable_key)),
+                        }
+                        for variable_key in action.entities
+                    ]
+                )
+            else:
+                requests.extend(
+                    [
+                        {
+                            "method": MAP_BULK_ACTION_TO_AUTH_METHOD[action.action],
+                            "details": VariableDetails(key=variable.key),
+                        }
+                        for variable in action.entities
+                    ]
+                )
```

Comment:
Would it be better to simplify as the following? 

```suggestion
        requests: list[IsAuthorizedVariableRequest] = [
            {
                "method": MAP_BULK_ACTION_TO_AUTH_METHOD[action.action],
                "details": VariableDetails(
                    key=cast(
                        "str",
                        entity if action.action == BulkAction.DELETE else entity.key
                    )
                ),
            }
            for entity in action.entities
        ]
    )
```

Reply:
True, I like it better :)

Reply:
Updated